pysteps/nowcasts/steps.py

```diff
@@ -33,6 +33,863 @@
+        self.precip_cascades = [
+            self.precip_cascades[i][-self.ar_order :]
+            for i in range(self.n_cascade_levels)
+        ]
+
+        # Stack the cascades into a list containing all ensemble members
+        self.precip_cascades = [
+            [self.precip_cascades[j].copy() for j in range(self.n_cascade_levels)]
+            for _ in range(self.n_ens_members)
+        ]
+
+        # Initialize random generators if noise_method is provided
+        if self.noise_method is not None:
+            self.randgen_prec = []
+            self.randgen_motion = []
+
+            for _ in range(self.n_ens_members):
+                # Create random state for precipitation noise generator
+                rs = np.random.RandomState(self.seed)
+                self.randgen_prec.append(rs)
+                self.seed = rs.randint(0, high=int(1e9))  # Update seed after generating
+
+                # Create random state for motion perturbations generator
+                rs = np.random.RandomState(self.seed)
+                self.randgen_motion.append(rs)
+                self.seed = rs.randint(0, high=int(1e9))  # Update seed after generating
+        else:
+            self.randgen_prec = None
+            self.randgen_motion = None
+        print("AR model and noise applied to precipitation cascades.")
+
+    def _initialize_velocity_perturbations(self):
+        """
+        Initialize the velocity perturbators for each ensemble member if the velocity
+        perturbation method is specified.
+        """
+        if self.vel_pert_method is not None:
+            init_vel_noise, generate_vel_noise = noise.get_method(self.vel_pert_method)
+
+            self.velocity_perturbations = []
+            for j in range(self.n_ens_members):
+                kwargs = {
+                    "randstate": self.randgen_motion[j],
+                    "p_par": self.vel_pert_kwargs.get("p_par", self.vp_par),
+                    "p_perp": self.vel_pert_kwargs.get("p_perp", self.vp_perp),
+                }
+                vp = init_vel_noise(
+                    self.velocity, 1.0 / self.kmperpixel, self.timestep, **kwargs
+                )
+                self.velocity_perturbations.append(
+                    lambda t, vp=vp: generate_vel_noise(vp, t * self.timestep)
+                )
+        else:
+            self.velocity_perturbations = None
+        print("Velocity perturbations initialized successfully.")
+
+    def _initialize_precipitation_mask(self):
+        """
+        Initialize the precipitation mask and handle different mask methods (sprog, incremental).
+        """
+        self.precip_forecast = [[] for _ in range(self.n_ens_members)]
+
+        if self.probmatching_method == "mean":
+            self.mu_0 = np.mean(
+                self.precip[-1, :, :][self.precip[-1, :, :] >= self.precip_thr]
+            )
+        else:
+            self.mu_0 = None
+
+        self.precip_m = None
```

Comment:
Should this be:
```suggestion
        self.precip_mask = None
```
?